grakn-graph/src/main/java/ai/grakn/graph/internal/TxCache.java

```diff
@@ -386,6 +386,11 @@ Json getFormattedLog(){
     void closeTx(String closedReason){
         isTxOpen = false;
         this.closedReason = closedReason;
+
+        //Clear Concept Caches
+        conceptCache.values().forEach(concept -> ((ContainsTxCache) concept).txCacheClear());
```

Comment:
Since we know this cast is safe, we can add one of your `ContainsTxCache.from(Concept)` methods.